driver-core/src/main/java/com/datastax/driver/core/TableMetadata.java

```diff
@@ -73,6 +73,10 @@ private TableMetadata(KeyspaceMetadata keyspace,
     static TableMetadata build(KeyspaceMetadata ksm, Row row, Map<String, ColumnMetadata.Raw> rawCols, List<Row> indexRows, String nameColumn, VersionNumber cassandraVersion, Cluster cluster) {
 
         String name = row.getString(nameColumn);
+        if(ksm.isVirtual()){
+            return  new TableMetadata(ksm, name, new UUID(0L, 0L), Collections.<ColumnMetadata>emptyList(), Collections.<ColumnMetadata>emptyList(),
+                    Collections.<String, ColumnMetadata>emptyMap(), Collections.<String, IndexMetadata>emptyMap(), null, Collections.<ClusteringOrder>emptyList(), cassandraVersion);
+        }
```

Comment:
So a virtual table's metadata never has columns? Shouldn't we expose the contents of `system_virtual_schema.columns` here?